mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,462 @@
+            for possible key-value pairs.
+        logging_interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        log_checkpoint_metadata (bool): Log the evaluation metrics computed
+            on the validation data with the checkpoint, along with current
+            epoch as a metadata to that checkpoint.
+            Default: True
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+    """
+
+    def __init__(self,
+                 wandb_init_kwargs=None,
+                 interval=10,
+                 log_checkpoint=False,
+                 log_checkpoint_metadata=False,
+                 num_eval_images=100,
+                 **kwargs):
+        super(WandbLogger, self).__init__(wandb_init_kwargs, interval,
+                                          **kwargs)
+
+        self.log_checkpoint = log_checkpoint
+        self.log_checkpoint_metadata = log_checkpoint_metadata
+        self.num_eval_images = num_eval_images
+        self.log_eval_metrics = True
+        self.best_score = 0
+        self.val_step = 0
+
+    @master_only
+    def before_run(self, runner):
+        super(WandbLogger, self).before_run(runner)
+        self.cfg = self.wandb.config
+        # Check if configuration is passed to wandb.
+        if len(dict(self.cfg)) == 0:
+            warnings.warn(
+                'To log mmdetection Config, '
+                'pass it to init_kwargs of WandbLogger.', UserWarning)
+
+        # Check if EvalHook and CheckpointHook are available.
+        for hook in runner.hooks:
+            if isinstance(hook, CheckpointHook):
+                self.ckpt_hook = hook
+            if isinstance(hook, (EvalHook, DistEvalHook)):
+                self.eval_hook = hook
+
+        # If CheckpointHook is not available turn off log_checkpoint.
+        try:
+            self._check_priority(self.ckpt_hook)
+        except AttributeError:
+            self.log_checkpoint = False
+            warnings.warn('To use log_checkpoint turn use '
+                          'CheckpointHook.', UserWarning)
+
+        # If EvalHook/DistEvalHook is not present set
+        # num_eval_images to zero.
+        try:
+            self.val_dataloader = self.eval_hook.dataloader
+            self.val_dataset = self.val_dataloader.dataset
+            self._check_priority(self.eval_hook)
+        except AttributeError:
+            self.num_eval_images = 0
+            self.log_checkpoint_metadata = False
+            self.log_eval_metrics = False
+            warnings.warn(
+                'To log num_eval_images turn validate '
+                'to True in train_detector.', UserWarning)
```

Comment:
The logger hooks usually have lower priority than eval and checkpoint hooks. We can only check the AttributeError.

Reply:
Thus `self._check_priority(self.eval_hook)` is not required. Does it mean that if the priority of the logger will be more than eval hook, it will raise AttributeError?